src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -718,6 +718,17 @@ private bool AreNamedTypesEqual(NamedTypeSymbol type, NamedTypeSymbol other)
                 // TODO: Test with overloads (from PE base class?) that have modifiers.
                 Debug.Assert(!type.HasTypeArgumentsCustomModifiers);
                 Debug.Assert(!other.HasTypeArgumentsCustomModifiers);
+
+                if (type.IsTupleType)
+                {
+                    if (!other.IsTupleType)
+                    {
+                        return false;
+                    }
+
+                    type.TupleElementTypes.SequenceEqual(other.TupleElementTypes, AreTypesEqual);
```

Comment:
Ok. With the change above (line 492), I don't think `AreNamedTypesEqual` can be called with a tuple type (the type should be unwrapped into a `ValueTuple` before getting here).
Should I leave this for safety, or maybe put an assert?